In [1]:
# Load pandas library
import pandas as pd

In [2]:
# identify the webpage by its url                                                                                                               
url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'

# read the first table into a dataframe
df = pd.read_html(url)[0]
df

Postal Code           Borough  \
0           M1A      Not assigned   
1           M2A      Not assigned   
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
..          ...               ...   
175         M5Z      Not assigned   
176         M6Z      Not assigned   
177         M7Z      Not assigned   
178         M8Z         Etobicoke   
179         M9Z      Not assigned   

                                         Neighbourhood  
0                                         Not assigned  
1                                         Not assigned  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
..                                                 ...  
175                                       Not assigned  
176                                       Not assigned  
177                                       Not assigned  
178  Mimico NW, The Queensway West, South of Bloor,...  
179                                       Not assigned  

[180 rows x 3 columns]

In [15]:
# drop rows with no assigned borough                                                                                                           
df_borough = df[df.Borough != 'Not assigned']
df_borough.shape

(103, 3)

The final dataset df_borough has 103 rows and 3 columns after removing not assigned boroughs and ensuring that no duplicate row with the same postal code.